14.0 change tg lib

mentalko opened the pull request.

Commit:
:sparkles: use another telegram lib

closes #295
refactoring of the code

yelizariev commented on sync_telegram/doc/changelog.rst:
```diff
@@ -1,3 +1,9 @@
+`5.0.0`
+-------
+
+- **Improvement:** moving to another telegram library
```
```suggestion
- **Improvement:** use another telegram library
```

yelizariev commented on sync_telegram/models/sync_project.py:
```diff
@@ -143,14 +131,14 @@ def parse_data(data):
+            "multi_livechat": multi_livechat_context,
+            "MAX_SIZE_IMAGE": MAX_SIZE_IMAGE,
+            "MAX_SIZE_DOCUMENT": MAX_SIZE_DOCUMENT,
+            "MAX_SIZE_TO_DOWNLOAD": MAX_SIZE_TO_DOWNLOAD,
```
Зачем убрал из словаря "телеграм"?

yelizariev commented on sync_telegram/models/sync_project.py:
```diff
@@ -62,76 +60,66 @@ def _html_sanitize_telegram(html):
         def setWebhook(*args, **kwargs):
             log_transmission("Telegram->setWebhook", json.dumps([args, kwargs]))
-            bot.setWebhook(*args, **kwargs)
+            bot.set_webhook(*args, **kwargs)
```
Здесь тоже можно  декоратор применить

yelizariev commented on sync_telegram/models/sync_project.py:
```diff
@@ -62,76 +60,66 @@ def _html_sanitize_telegram(html):
+            return [file_path.split("/")[-1], content]
 
+        @LogExternalQuery(
+            "Telegram->The user's photo has been added to the partner record.",
```
В этом параметре нужно дать название внешнему запросу. Не нужно предложение целое писать

yelizariev commented on sync_telegram/data/sync_project_data.xml:
```diff
@@ -10,21 +10,17 @@
-
+TG_PARTNER="TG_PARTNER"
+TG_CHANNEL="TG_CHANNEL"
+TG_THREAD="TG_THREAD"
```
Так ты все существующие установки ломаешь. Лучше не ломать ссылки, если можно не ломать

mentalko commented:
тест ругается на pyTelegramBotAPI

yelizariev commented on requirements.txt:
```diff
@@ -1,6 +1,6 @@
 bravado_core
 facebook_business
-python-telegram-bot
+pyTelegramBotAPI
```
Демо проект в Sync Studio тоже придется переделать, т.к. имя пакета в питоне одинаковое

Commit:
:heart_eyes: sync: use another telegram library in demo project

removed `html2plaintext` method from sync_project_demo.py because its already has in global eval

yelizariev commented on sync/data/sync_project_telegram_demo.xml:
```diff
@@ -85,11 +87,12 @@ def handle_webhook(httprequest):
     update = telegram.parse_data(data)
     message = update.message or update.edited_message
     is_edit = bool(update.edited_message)
-    user_ref = "%s@telegram" % message.from_user.id
```
А зачем ты меняешь эту часть?

mentalko replied:
логичнее использовать преимущества линков

yelizariev replied:
Логичнее, но это демо, которое показывает другие стороны. Например, можно понять как модуль работает не погружаясь сразу в линки

yelizariev commented on sync/data/sync_project_telegram_demo.xml:
```diff
@@ -120,14 +123,13 @@ def handle_db(records):
         partner = env["res.partner"].browse(r.res_id)
-        user_ref = partner.ref or ""
-        if not user_ref.endswith("@telegram"):
+        log(str(partner), LOG_DEBUG)
```
Логи нужно понятные печатать или убрать. Как сейчас сделано будет прост партер печататься и нужно догадываться к чему это отностися

yelizariev commented on sync/data/sync_project_telegram_demo.xml:
```diff
@@ -120,14 +123,13 @@ def handle_db(records):
         telegram_message_html = html_sanitize_telegram(r.body)
         telegram_message_html = telegram_message_html.replace("/telegram", "")
         telegram.sendMessage(telegram_user_id, telegram_message_html)
-
```
Я так понимаю в xml можно вообще ничего не менять -- не зависит от библиотеки. Если ты хочешь что-то еще изменить в демо проекте, то нужно обосновать и указать изменения в changelog.rst

mentalko replied:
логичнее использовать преимущества линков

yelizariev commented on sync/doc/changelog.rst:
```diff
@@ -1,3 +1,8 @@
+`5.1.0`
+-------
+
+- **Improvement:** use another telegram library in Telegram demo project
```
```suggestion
- **Improvement:** use light telegram library in Telegram demo project (pyTelegramBotAPI)
```

yelizariev commented on sync/models/sync_project_demo.py:
```diff
@@ -18,12 +20,6 @@
 
 _logger = logging.getLogger(__name__)
 
-try:
-    # https://github.com/python-telegram-bot/python-telegram-bot
-    from telegram import Bot, Update  # pylint: disable=missing-manifest-dependency
-except (ImportError, IOError) as err:
-    _logger.debug(err)
```
Нужно сохранить трай блок. Он не просто так здесь. Иначе работать не будет пока телеграм библиотеку не установишь, которая далеко не всем нужна

yelizariev commented on sync/models/sync_project_demo.py:
```diff
@@ -121,7 +115,6 @@ def parse_data(data):
 
         return {
             "telegram": telegram,
-            "html2plaintext": html2plaintext,
```
Обоснуй в описании комита

mentalko replied:
этот метод из тулзов убрали вот и не нужен он теперь

yelizariev replied:
Да, но это нужно в описание комита добавить

yelizariev commented on sync/doc/changelog.rst:
```diff
@@ -1,3 +1,11 @@
+-------
+
+- **Improvement:** use light telegram library in Telegram demo project (pyTelegramBotAPI)
+- **Improvement:** add an example of using links to the Telegram demo project
```
Не надо)
Ссылки много еще где можно посмотреть. Тут специально пример сделан без ссылок

yelizariev commented on sync/doc/changelog.rst:
```diff
@@ -1,3 +1,11 @@
+
+- **Improvement:** use light telegram library in Telegram demo project (pyTelegramBotAPI)
+- **Improvement:** add an example of using links to the Telegram demo project
+- **Fix:** remove `html2plaintext` method from eval context
```
Это только запутает клиента. Ты написал "удаляем". А он не удаляется. Этот метод доступен к использованию. Он просто через глобально везде добавляется, не нужно в контекте проекта его еще раз добавлять. Удали эту строчку здесь. А в описании комита напиши почему удалил код

yelizariev commented on sync/data/sync_project_telegram_demo.xml:
```diff
@@ -12,13 +12,10 @@
 def user2name(user):
     if user.username:
         return '@%s' % (user.username)
-
```
Может не будешь пустые строчки удалять?

yelizariev commented on sync_telegram/doc/changelog.rst:
```diff
@@ -1,3 +1,9 @@
+-------
+
+- **Improvement:** use another telegram library
+- **Improvement:** code refactoring
```
Про рефакторинг в описании комита еще укажи

yelizariev reviewed (approved):
Ну вроде норм, если работает
